android/jni/app-android.cpp

```diff
@@ -612,16 +620,23 @@ extern "C" void Java_org_ppsspp_ppsspp_NativeRenderer_displayInit(JNIEnv * env,
 	if (javaGL && !graphicsContext) {
 		graphicsContext = new AndroidJavaEGLGraphicsContext();
 	}
-	ILOG("NativeApp.displayInit() (graphicsContext=%p)", graphicsContext);
 
-	if (!renderer_inited) {
+	if (renderer_inited) {
+		ILOG("NativeApp.displayInit() restoring");
+		NativeDeviceLost();
+		NativeShutdownGraphics();
+		NativeDeviceRestore();
+		NativeInitGraphics(graphicsContext);
```

Comment:
Hm.  `g_gameInfoCache = new GameInfoCache();` is in `NativeInitGraphics`.  Shouldn't we init before we restore?  Something might try to access g_gameInfoCache during restore?

-[Unknown]